core/src/test/java/com/sequenceiq/cloudbreak/converter/v4/environment/network/SubnetSelectorTest.java

```diff
@@ -71,39 +80,98 @@ public void testChooseEndpointGatewaySubnetFromEnvironmentSubnetst() {
             "key2", getPublicCloudSubnet(PUBLIC_ID_1, AZ_1)
         ));
         source.setPublicEndpointAccessGateway(PublicEndpointAccessGateway.ENABLED);
+        when(entitlementService.endpointGatewayValidationEnabled(any())).thenReturn(false);
 
         Optional<CloudSubnet> subnet =
-            underTest.chooseSubnetForEndpointGateway(source, PRIVATE_ID_1);
+            ThreadBasedUserCrnProvider.doAs(USER_CRN, () -> underTest.chooseSubnetForEndpointGateway(source, PRIVATE_ID_1));
 
         assert subnet.isPresent();
         assertEquals(PUBLIC_ID_1, subnet.get().getId());
     }
 
     @Test
-    public void testChooseEndpointGatewaySubnetNoPublicEndpointSubnets() {
+    public void testChooseEndpointGatewaySubnetNoPublicEndpointSubnetsEntitlementEnabled() {
```

Comment:
Shouldn't the test names be updated with new entitlement(`entitlementService.endpointGatewayValidationEnabled`)?

Reply:
That would be more clear. Thanks for the suggestion.